src/test/java/com/amihaiemil/camel/ScalarTest.java

```diff
@@ -82,4 +84,25 @@ public void orphanForbidden() {
             new ArrayList<AbstractNode>(), "orphan"
         );
     }
+
+    /**
+     * Make sure that equals and hash code are reflexive,
+     * symmetric, and transitive.
+     */
+    @Test
+    public void testEqualsAndHashCode() {
+        final String firstValue = "test scalar value #1";
+        final String secondValue = "test scalar value #2";
+        final Scalar<String> firstScalar = new Scalar<String>(
+            Arrays.asList(Mockito.mock(AbstractNode.class)),
+            firstValue
+        );
+        final Scalar<String> secondScalar = new Scalar<String>(
+            Arrays.asList(Mockito.mock(AbstractNode.class)),
+            secondValue
+        );
+        Assert.assertTrue(firstScalar.equals(secondScalar)
+            && secondScalar.equals(firstScalar));
+        Assert.assertTrue(firstScalar.hashCode() == secondScalar.hashCode());
```

Comment:
@salikjan the assert methods from class Assert are ok (it's also what I use a lot in other projects). But with this projects, I would like to start using Matchers. They are more robust and offer more descriptive failure messages. Can you look in the test above, how I did it? 

E.g. for the assert equals above, it would look something like this:

```
MatcherAssert.assertThat(
    firstScalar, Matchers.equals(secondScalar)
);
```
But I'm not sure, you'll have to play a bit with it :D